### PR TITLE
i18n will double-quote YAML strings with '#' in them

### PR DESCRIPTION
**Pull request by daynew**
Adds double quotes around strings which have an '#'  in them (markdown formatting); otherwise, YAML parsers might treat the string as a comment.

Ideally we would add double quotes around all Strings, but testing revealed that Crowdin's YAML parser has some bugs and can't parse some specific use cases. Fortunately, none of our markdown strings seem to hit those edge cases.

## Testing story
* Ran a full i18n sync on the `i18n-dev` server

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: bin/i18n/i18n_script_utils.rb [hunk 1]
```diff
@@ -55,10 +55,17 @@ def self.to_crowdin_yaml(data)
...
     # Make sure we treat the strings 'y' and 'n' as strings, and not bools
     yaml_bool = /^(?:y|Y|n|N)$/
+    # Make sure we use the right format for strings with '#' in them, otherwise
+    # YAML parsers might treat part of the string as a YAML comment.
+    octothorpe = /#/
     ast.grep(Psych::Nodes::Scalar).each do |node|
       if yaml_bool.match node.value
         node.plain = false
         node.quoted = true
+      elsif octothorpe.match node.value
```
Comment (annaxuphoto) [inline bin/i18n/i18n_script_utils.rb:65]: Do we need a separate block for '#' strings if we're doing the same thing to them as the booleans?

Comment (daynew) [reply bin/i18n/i18n_script_utils.rb:65]: I agree that they could probably be merged. For this PR, I wanted to focus on minimizing changes since we have been struggling to get an i18n sync out.
